src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj

```diff
@@ -19,8 +19,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataToolsetVersion)" />
-    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonToolsetVersion)" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Reflection.Metadata\src\System.Reflection.Metadata.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Text.Json\src\System.Text.Json.csproj" />
```

Comment:
These references need to not move newer than MSBuild as this library is consumed by MSBuild tasks in the SDK (including for the .NET Framework build of the SDK targets). We can't reference live builds as a result.

Reply:
The SDK is already on the live version of `System.Text.Json` for its tasks, but it seems it is not on the live version of `System.Reflection.Metadata`. We have another library (Microsoft.Extensions.DependencyModel) that SDK consumes with live JSON as well.  I only needed to update to live STJ to fix the audit warning so I could undo the SRM change, but it feels wrong.

I'll file a follow up issue in the SDK for that.

Reply:
> The SDK is already on the live version of System.Text.Json for its tasks

Would you mind elaborating? Last time I looked, NETFramework tasks in the SDK were using an 8.0.x version of STJ because MSBuild's binding redirects required that.

Reply:
The SDK tasks themselves ('Microsoft.NET.Build.Tasks.dll`) reference the latest.  Because it's an exact match they don't need the redirects.  Other assemblies loaded by the SDK do reference lower versions, and they rely on MSBuild's redirects, but they don't exchange types with the SDK tasks.  The SDK also ships the latest STJ along side its tasks.
For example:
```
"C:\Program Files\dotnet\sdk\9.0.100-preview.7.24407.12\Sdks\Microsoft.NET.Sdk\tools\net472\System.Text.Json.dll" => 9.0.0.0
"C:\Program Files\dotnet\sdk\8.0.400\Sdks\Microsoft.NET.Sdk\tools\net472\System.Text.Json.dll" => 8.0.0.4
```

The SDK's MSBuild SDK resolver does need to stay on the VS copy of STJ because it doesn't carry a copy next to itself.
CC @marcpopMSFT